### PR TITLE
mtd/smartfs: Fix compilation warning

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Fix compilation warning
## Impact
Improve compilation
## Testing
sim:smartfs

### CODE REVIEW

File: drivers/mtd/smart.c [hunk 1]
```diff
@@ -856,7 +856,8 @@ static ssize_t smart_reload(struct smart_struct_s *dev, FAR uint8_t *buffer,
 
   /* Read the full erase block into the buffer */
 
-  finfo("Read %d blocks starting at block %d\n", mtdblocks, mtdstartblock);
+  finfo("Read %" PRIdOFF "blocks starting at block %" PRIdOFF "\n",
```
Comment (gustavonihei) [inline drivers/mtd/smart.c:859]:
```suggestion
  finfo("Read %" PRIdOFF " blocks starting at block %" PRIdOFF "\n",
```
Missing whitespace after the offset value.

Comment (acassis) [reply drivers/mtd/smart.c:859]: Nice catch! Thank you!

File: drivers/mtd/mtd_partition.c [hunk 2]
```diff
@@ -579,7 +579,7 @@ static ssize_t part_procfs_read(FAR struct file *filep, FAR char *buffer,
                   MTDIOC_GEOMETRY, (unsigned long)((uintptr_t)&geo));
           if (ret < 0)
             {
-              ferr("ERROR: mtd->ioctl failed: %d\n", ret);
+              ferr("ERROR: mtd->ioctl failed: %" PRId32 "\n", ret);
```
Comment (gustavonihei) [inline drivers/mtd/mtd_partition.c:582]:
```suggestion
              ferr("ERROR: mtd->ioctl failed: %zd\n", ret);
```
`ssize_t` (as well as `size_t`) are implementation-dependent, so we should not use a fixed-length specifier for them.
According to the C standard, the correct specifier for size types is **z**.

**Review by gustavonihei — CHANGES_REQUESTED** [review]
Dropping the approval for now due to latest changes.

`PRIdOFF` and `PRIuOFF` were added for printing values of `off_t` type, not for size types.

**Comment by acassis** [comment]
Done!

File: drivers/mtd/smart.c [hunk 3]
```diff
@@ -4560,7 +4561,7 @@ static int smart_writesector(FAR struct smart_struct_s *dev,
       /* Subtract dev->minwearlevel from all wear levels */
 
       offset = dev->minwearlevel;
-      finfo("Reducing wear level bits by %d\n", offset);
+      finfo("Reducing wear level bits by %zd\n", offset);
```
Comment (gustavonihei) [inline drivers/mtd/smart.c:4564]:
```suggestion
      finfo("Reducing wear level bits by %zu\n", offset);
```
`offset` here is defined as unsigned size type.

File: drivers/mtd/smart.c [hunk 4]
```diff
@@ -3213,7 +3214,7 @@ static inline int smart_llformat(FAR struct smart_struct_s *dev,
     {
       /* The block is not empty!!  What to do? */
 
-      ferr("ERROR: Write block 0 failed: %d.\n", wrcount);
+      ferr("ERROR: Write block 0 failed: %zd.\n", wrcount);
```
Comment (gustavonihei) [inline drivers/mtd/smart.c:3217]:
```suggestion
      ferr("ERROR: Write block 0 failed: %zu.\n", wrcount);
```
`wrcount` is unsigned.

**Review by gustavonihei — APPROVED** [review]

**Pull request merged**